aip/general/0203.md

```diff
@@ -161,6 +161,29 @@ the user's behalf.
 A resource with an unordered list **may** return the list in a stable order, or
 **may** return the list in a randomized, unstable order.
 
+## Backwards compatibility
+
+Adding or changing `google.api.field_behavior` values can represent a semantic
+change in the API that is perceived as incompatible for existing clients. The
+following are examples of backwards incompatible changes with
+`google.api.field_behavior`:
+
+* Adding `REQUIRED` to an existing field previously considered `OPTIONAL`
+(implicitly or otherwise).
+* Adding a new field annotated as `REQUIRED` to an existing request message.
+* Adding `OUTPUT_ONLY` to an existing field previously accepted as input
+* Removing `OUTPUT_ONLY` from an existing field previously ignored as input
```

Comment:
Why would changing `OUTPUT_ONLY` to `OPTIONAL` (this is what removing it would do) be semantically breaking?

Example: imagine an API generated some IDs for resources being created (e.g. a link shortener) but now allows to optionally provide the ID (proposed short link). Would this change be considered breaking?

Reply:
Take a look through [this thread](https://github.com/aip-dev/google.aip.dev/pull/1172/#discussion_r1269972428) where we discuss it a bit more. It is potentially hazardous for situations where the field:
* was output_only
* potentially unset in a patch because of the output_only nature
* patch used updateMask=*
* no longer output_only means that the field will be cleared unintentionally

Basically suddenly accepting something as input when previously it was ignored presents a potentially sneaky change in behavior that. This is the conservative perspective for sure, but I think it is one worth taking.